Mage/src/main/java/mage/abilities/common/DiesThisOrAnotherCreatureTriggeredAbility.java

```diff
@@ -44,18 +44,8 @@ public boolean checkEventType(GameEvent event, Game game) {
 
     @Override
     public boolean isInUseableZone(Game game, MageObject source, GameEvent event) {
-        Permanent sourcePermanent = null;
-        if (game.getState().getZone(getSourceId()) == Zone.BATTLEFIELD) {
-            sourcePermanent = game.getPermanent(getSourceId());
-        } else {
-            if (game.getShortLivingLKI(getSourceId(), Zone.BATTLEFIELD)) {
-                sourcePermanent = (Permanent) game.getLastKnownInformation(getSourceId(), Zone.BATTLEFIELD);
-            }
-        }
-        if (sourcePermanent == null) {
-            return false;
-        }
-        return hasSourceObjectAbility(game, sourcePermanent, event);
+        Permanent sourcePermanent = game.getPermanentOrLKIBattlefield(getSourceId());
```

Comment:
That radical changes with removes of `getShortLivingLKI` need more research and tests. That's code uses in many another abilities too. Maybe @LevelX2 can hint on that.